src/components/tasks/AssignTasks.js

```diff
@@ -0,0 +1,113 @@
+import { makeStyles } from '@material-ui/core/styles';
+import { Divider, Grid, Button, Typography, GridList, GridListTile } from '@material-ui/core';
+import Modal from '@material-ui/core/Modal';
+import PropTypes from 'prop-types';
+import Backdrop from '@material-ui/core/Backdrop';
+import Fade from '@material-ui/core/Fade';
+import { Task } from './crud/Task'
+import { tasksApi } from '../../utils/apis/tasksAPI'
+import { useDevPassword } from '../../utils/hooks/useDevPassword'
+
+const useStyles = makeStyles((theme) => ({
+    modal: {
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+      },
+    paper: {
+      position: 'absolute',
+      width: '50%',
+      backgroundColor: theme.palette.background.paper,
+      border: '2px solid #000',
+      boxShadow: theme.shadows[5],
+      padding: theme.spacing(2, 4, 3),
+    },
+  }));
+
+  function getModalStyle() {
+    const top = 50;
+    const left = 50;
+  
+    return {
+      top: `${top}%`,
+      left: `${left}%`,
+      transform: `translate(-${top}%, -${left}%)`,
+    };
+  }
+
+
+export const AssignTasks = ({ modalOptions, alreadyAssignedTasks, organizationTasks, setReload, assignKey, assignUUIDValue }) => {
+
+    const classes = useStyles()
+    const {getEncoded} = useDevPassword()
+
+    const onAssign = (task) => {
+        const tasksIds = alreadyAssignedTasks.map(assignedTask => assignedTask.id)
+        tasksApi.assignTasks([...tasksIds, task.id], getEncoded(), setReload, assignKey, assignUUIDValue)
+    }
+
+    const onDeassign = (task) => {
+        const newTasksIds = alreadyAssignedTasks.filter(assignedTask => assignedTask.id !== task.id).map(task => task.id)
+        tasksApi.assignTasks(newTasksIds, getEncoded(), setReload, assignKey, assignUUIDValue)
+    }
+
+    return (
+        <Modal
+            open={modalOptions.open}
+            onClose={modalOptions.onClose}
+            className={classes.modal}
+            closeAfterTransition
+            BackdropComponent={Backdrop}
+            BackdropProps={{
+                timeout: 500,
+            }}
+            style={{zIndex: 1000}}
+        >
+            <Fade in={modalOptions.open}>
+                <div className={classes.paper} style={getModalStyle()}>
+                    
+                    <Section title="Tasks you've already assigned:" tasks={alreadyAssignedTasks} empty="No one have assign any tasks for this application" actionTitle="Deassign" action={onDeassign} />
```

Comment:
`has assigned`*

Reply:
`tasks you have already assigned` - druga osoba ma `have`, nie `has`